src/libs/ajax/User.ts

```diff
@@ -0,0 +1,168 @@
+import * as fp from 'lodash/fp';
+import { cloneDeep, flow, unset } from 'lodash/fp';
+import { Config } from '../config';
+import axios from 'axios';
+import { getApiUrl, fetchOk, fetchAny } from '../ajax';
+import { CreateDuosUserResponse, DuosUserResponse, UpdateDuosUserResponse } from 'src/types/responseTypes';
+import { CreateDuosUserRequest, UpdateDuosUserRequestV1, UpdateDuosUserRequestV2 } from 'src/types/requestTypes';
+import { AcknowledgementMap, ApprovedDataset, Dataset, DuosUser, SimplifiedDuosUser } from 'src/types/model';
+
+export const User = {
+  getMe: async (): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/me`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  getById: async (id: number): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/${id}`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  list: async (roleName: 'Admin' | 'SigningOfficial'): Promise<DuosUserResponse[]> => {
+    const url = `${await getApiUrl()}/api/user/role/${roleName}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return res.json();
+  },
+
+  create: async (request: CreateDuosUserRequest): Promise<CreateDuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/dacuser`;
+    try {
+      const res = await fetchOk(
+        url,
+        fp.mergeAll([
+          Config.authOpts(),
+          Config.jsonBody(request),
+          { method: 'POST' },
+        ])
+      );
+      if (res.ok) {
+        return res.json();
+      }
```

Comment:
Wanted to make a callout here. This previously read `return res.json` without the parenthesis and Typescript helped me catch that. There is only one place this method is used and it is being used as a truthy value so I don't believe this should change any behavior.

Regardless, I just wanted to make reviewers aware since it is not clear at all from the diff.

Reply:
I think this is fine, but it might be redundant if `fetchOk` fails on a non-OK response. There could be descendant code that also handles the non-OK case in a particular manner, for example, showing a custom error response depending on what `res.json()` does or does not contain.

Reply:
Just to clarify - I don't see any issues with moving from `res.json` -> `res.json()`

Reply:
I think ultimately, we should change all these `fetchOk`s to use axios instead. I think it would be wise to have a consistent way of making these requests and if we decide to do any handling of errors, they should happen outside of these ajax calls